dts/bindings/adc/adc-comparator.yaml

```diff
@@ -0,0 +1,14 @@
+# Copyright (c) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    Comparator will continuously monitor signal through ADC channel and compare
+    its result with defined threshold value. When comparison criterion is met,
+    event will be triggered.
+
+properties:
+    io-channels:
+      type: phandle-array
+      required: true
+      description: |
+        ADC channel that will perform measurement.
```

Comment:
I am not sure this is generic enough as a base binding. I would suggest just folding it into the `nuvoton,adc-cmp.yaml` binding until we have more users of this.

Reply:
The ongoing work that wants to use this driver is also implementing equivalent driver support for it8xxx2; is one more user enough?

Reply:
OK, done, I also moved dts/bindings/adc/nuvoton,adc-cmp.yaml -> dts/bindings/sensor/nuvoton,adc-cmp.yaml